solr/core/src/java/org/apache/solr/update/processor/AtomicUpdateDocumentMerger.java

```diff
@@ -143,6 +147,15 @@ public SolrInputDocument merge(final SolrInputDocument fromDoc, SolrInputDocumen
     return toDoc;
   }
 
+  private static String getID(SolrInputDocument doc, IndexSchema schema) {
+    String id = "";
```

Comment:
can we default to `(unknown id)` otherwise the error message will look weird I think.

Reply:
I'm thinking to rephrase the above error message to something like so that it is better than the previous msg. If the id is not known then, I think maybe it is better not to send anything related id, wdyt?
```
"Error:" + getID(toDoc, schema) + "Unknown operation for the an atomic update : " + key;
```